plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2115,6 +2120,28 @@ private static boolean fieldExists(StructType structType, String fieldName)
         return false;
     }
 
+    private static void checkDuplicatedFieldNames(io.trino.spi.type.Type type)
+    {
+        if (type instanceof ArrayType arrayType) {
+            checkDuplicatedFieldNames(arrayType.getElementType());
+        }
+        else if (type instanceof MapType mapType) {
+            checkDuplicatedFieldNames(mapType.getKeyType());
+            checkDuplicatedFieldNames(mapType.getValueType());
+        }
+        if (type instanceof RowType rowType) {
+            Set<String> fieldNames = new HashSet<>();
+            for (RowType.Field field : rowType.getFields()) {
+                field.getName().ifPresent(name -> {
```

Comment:
`field.getName()` must be present. anonymous rows are not supported

Reply:
I avoided throwing an exception because an empty field name is checked in a different place:
https://github.com/trinodb/trino/blob/abc57cc2921577ece33624abc5920a9bf99e5ab7/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TypeConverter.java#L182-L199

Reply:
can we do both checks in one place?